mindmeld/components/dialogue.py

```diff
@@ -876,20 +877,21 @@ def _validate(self, request, slot):
 
     def _initial_fill(self, request):
         """Performs the first pass and fills the entity form with entity values available
-        in the initial query.
+        in the initial query. Bypassed if `clear_prev` is set to True.
 
         Args:
             request (Request): The request object.
         """
-        for entity in request.entities:
```

Comment:
To prevent this indentation sprawl, you could do this:
```
if self._clear_prev:
    return

....

```

Reply:
Reverted some changes, I don't think this is needed anymore.